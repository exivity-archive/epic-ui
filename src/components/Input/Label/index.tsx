import * as React from 'react'
import styled from 'styled-components';
import theme, { Theme } from '../../../theme'

const LabelWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  flex-wrap: wrap;
  max-width: 650px;
  margin: 25px 0;
`

const SubLabelWrapper = styled.div`
  display: flex;
  flex-direction: column;
  min-width: 250px;
`

const TopLabel = styled.span`
  margin-top: -4px;
  font-weight: 600;
  color: #444444;
`
interface ISublabelProps {
  	theme: Theme
}

const SubLabel = styled.div`
  font-size: small;
  color: ${(p: ISublabelProps) => p.theme.colors.gray};
  margin-top: -4px;
`

interface ILabelProps {
  name: string
  description?: string
}

const Label: React.FC<ILabelProps> = ({ name, description, children }) => (
  <LabelWrapper>
    <SubLabelWrapper>
      <TopLabel>{name}</TopLabel>
      <SubLabel theme={theme}>{description}</SubLabel>
    </SubLabelWrapper>
    {children}
  </ LabelWrapper>
)



export default Label