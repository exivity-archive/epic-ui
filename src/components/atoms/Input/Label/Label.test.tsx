import React from 'react'
import renderer from 'react-test-renderer'

import Label from '.'
import { shallow } from 'enzyme';

test('renders label', () => {
  const label = renderer.create(<Label name="test" />)
  expect(label).toMatchSnapshot()
})
