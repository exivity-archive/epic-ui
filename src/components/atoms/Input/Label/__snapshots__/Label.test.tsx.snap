// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders label 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  max-width: 650px;
  margin: 25px 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-width: 250px;
}

.c2 {
  font-weight: 600;
  color: #444444;
}

.c3 {
  font-size: small;
  color: #aaa;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
    >
      test
    </span>
    <div
      className="c3"
    />
  </div>
</div>
`;
